kernel/Kconfig

```diff
@@ -987,12 +987,11 @@ config SMP_BOOT_DELAY
 	  secondary CPUs at a later time.
 
 config MP_NUM_CPUS
-	int "Number of CPUs/cores"
+	int "Number of CPUs/cores [DEPRECATED]"
 	default MP_MAX_NUM_CPUS
 	range 1 12
```

Comment:
AFAIK you need to 

`select DEPRECATED`

Reply:
AFAICT this does not work well range types that are always defined.

Reply:
I bet if we looked at one of the previous PRs from two years ago that sought to remove this config, we would see the same observation.

I don't want to actually do that search, but will approve preemptively assuming that to be true 😁